akka-docs/src/main/paradox/persistence-typed.md

```diff
@@ -1,32 +1,179 @@
+  are stored or none of them are stored if there is an error
+* `Effect.none` no events are to be persisted, for example a read-only command
+* `Effect.unhandled` the command is unhandled (not supported) in current state
+
+External side effects can be performed after successful persist with the `andThen` function. 
+In the example below a reply is sent to the `replyTo`. Note that the new state after applying 
+the event is passed as parameter to the `andThen` function.
+
+### Event handler
+
+When an event has been persisted successfully the current state is updated by applying the 
+event to the current state with the `eventHandler` function. 
+
+The event handler returns the new state, which must be immutable so you return a new instance of the state. 
+The same event handler is also used when the entity is started up to recover its state from the stored events.
+
+
+## Basic example
+
+Command and event:
+
+Scala
+:  @@snip [PersistentActorCompileOnyTest.scala]($akka$/akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentActorCompileOnlyTest.scala) { #command }
+
+State is a List cnntaining all the events:
+
+Scala
+:  @@snip [PersistentActorCompileOnyTest.scala]($akka$/akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentActorCompileOnlyTest.scala) { #state }
+
+The command handler just persists the `Cmd` payload in an `Evt`:
+
+Scala
+:  @@snip [PersistentActorCompileOnyTest.scala]($akka$/akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentActorCompileOnlyTest.scala) { #command-handler }
+
+The event handler appends the event to the state. This is called after successfully
+persisting the event in the database:
+
+Scala
+:  @@snip [PersistentActorCompileOnyTest.scala]($akka$/akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentActorCompileOnlyTest.scala) { #event-handler }
+
+These are used to create a `PersistentBehavior`:
+
+Scala
+:  @@snip [PersistentActorCompileOnyTest.scala]($akka$/akka-persistence-typed/src/test/scala/akka/persistence/typed/scaladsl/PersistentActorCompileOnlyTest.scala) { #behavior }
+
+The behavior can then be run as with any normal typed actor as described in [typed actors documentation](actors-typed.md).
+
+## Larger example
+
+After processing a message an ordinary, non-persistent, typed actor returns the `Behavior` that is used 
+for next message. 
+
+As you can see in the above examples this is not supported by typed persistent actors. Instead, the state is 
+returned by `eventHandler`. The reason a new behavior can't be returned is that behavior is part of the actor's 
+state and must also carefully be reconstructed during recovery. If it would have been supported it would mean 
+that the behavior must be restored when replaying events and also encoded in the state anyway when snapshots are used. 
+That would be very prone to mistakes.
```

Comment:
, and thus is not allowed in Typed Persistence?

Reply:
+1